hack/dind

```diff
@@ -10,6 +10,11 @@ set -e
 
 # Usage: dind CMD [ARG...]
 
+if [ -n "$DOCKER_SYSTEMD" ]; then
```

Comment:
Wondering if this should be handled through a build-arg, so that `hack/dind` doesn't have to be modified, and an if/else construct in the dockerfile;

```
FROM foo AS systemd-true
COPY hack/dind-systemd hack/dind-systemd
ENTRYPOINT ["hack/dind-systemd"]

FROM foo AS systemd-false
COPY hack/dind-systemd hack/dind
ENTRYPOINT ["hack/dind"]


FROM systemd-${DOCKER_SYSTEMD} AS foo
```

Reply:
@cpuguy83 @tonistiigi WDYT?

Reply:
I like @thaJeztah's suggestion.

Reply:
updated